design/token.go

```diff
@@ -50,6 +50,45 @@ var _ = a.Resource("token", func() {
 		a.Response(d.BadRequest, JSONAPIErrors)
 		a.Response(d.InternalServerError, JSONAPIErrors)
 	})
+
+	a.Action("link", func() {
+		a.Routing(
+			a.POST("link"),
+		)
+		a.Payload(linkPayload)
+		a.Description("Link the user account to an external resource provider such as GitHub")
+		a.Response(d.SeeOther)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+	})
+
+	a.Action("callback", func() {
+		a.Routing(
+			a.GET("/callback"),
+		)
+		a.Params(func() {
+			a.Param("code", d.String, "Code provided by an external oauth2 resource provider")
+			a.Param("state", d.String, "State generated by the link request")
+			a.Required("code")
+			a.Required("state")
+		})
+		a.Description("Callback from an external oauth2 resource provider such as GitHub as part of user's account linking")
+		a.Response(d.TemporaryRedirect)
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+	})
+})
+
+var linkPayload = a.Type("LinkPayload", func() {
+	a.Attribute("for", d.String, "Resource we need to link accounts for", func() {
+		a.Example("https://github.com/somecoolrepo")
+	})
+	a.Attribute("token", d.String, "User's access token")
```

Comment:
refresh my memory please - why aren't we picking up from the header?

Reply:
Because we can't use AJAX to initiate linking. So we have to use window.location = ... (GET) or submit a form (POST). In both cases we can't add headers to the request. Form/Post is preferable because we don't want to pass tokens as part of URL.